api/v1alpha1/nodepool_types.go

```diff
@@ -59,7 +59,9 @@ type NodePoolSpec struct {
 	Release Release `json:"release"`
 
 	// IgnitionService defines how the MachineConfigServer service is published in the management cluster
-	IgnitionService ServicePublishingStrategy `json:"ignitionService"`
+	// +kubebuilder:validation:Optional
```

Comment:
Same thinking than in https://github.com/openshift/hypershift/pull/169/files#r618204517
IMO this field should be required and defaulted as in here. That's just UX.
Then the backend controller should not be doing any assumption, it should handle all input possibilities and so it should fail appropriately when the field does not exist.